libeos-updater-flatpak-installer/perform-flatpak-actions.c

```diff
@@ -397,10 +452,60 @@ check_if_flatpak_is_installed (FlatpakInstallation        *installation,
     }
   g_clear_error (&local_error);
 
-  *out_is_installed = (installed_ref != NULL);
-  g_message ("Flatpak described by ref %s is %s",
-             formatted_ref,
-             *out_is_installed ? "installed": "not installed");
+  if (match_remote)
+    {
+      if (installed_ref != NULL)
+        {
+          const gchar *installed_remote = flatpak_installed_ref_get_origin (installed_ref);
+          *out_is_installed = g_strcmp0 (installed_remote, action->ref->remote);
+        }
+      else
+        *out_is_installed = FALSE;
+      g_message ("Flatpak described by ref %s:%s is %s",
+                 action->ref->remote, formatted_ref,
+                 *out_is_installed ? "installed" : "not installed");
+    }
+  else
+    {
+      *out_is_installed = (installed_ref != NULL);
+      g_message ("Flatpak described by ref %s is %s",
+                 formatted_ref,
+                 *out_is_installed ? "installed" : "not installed");
+    }
+
+  return TRUE;
+}
+
+static gboolean
+check_if_ostree_ref_exists (FlatpakInstallation        *installation,
+                            EuuFlatpakRemoteRefAction  *action,
+                            gboolean                   *out_exists,
+                            GError                    **error)
+{
+  g_autoptr(GFile) installation_directory = flatpak_installation_get_path (installation);
+  g_autoptr(GFile) repo_directory = g_file_get_child (installation_directory, "repo");
+  g_autoptr(OstreeRepo) repo = ostree_repo_new (repo_directory);
+  g_autofree gchar *formatted_ref = NULL;
+  g_autofree gchar *refspec = NULL;
+  g_autofree gchar *rev = NULL;
+
+  if (!ostree_repo_open (repo, NULL, error))
+    return FALSE;
+
+  formatted_ref = flatpak_ref_format_ref (action->ref->ref);
+  refspec = g_strdup_printf ("%s:%s", action->ref->remote, formatted_ref);
+  if (!ostree_repo_resolve_rev_ext (repo,
```

Comment:
Should this be taking the collection ID into account too? Perhaps with `ostree_repo_resolve_collection_ref()`?